src/typechecker/FStar.TypeChecker.TcTerm.fst

```diff
@@ -3700,53 +3689,21 @@ and check_top_level_let env e =
                 BU.print1 "Let binding AFTER tcnorm: %s\n" (Print.term_to_string e1);
 
          (*
-          * AR: we now compute comp for the whole `let x = e1 in e2`, where e2 = ()
+          * AR: comp for the whole `let x = e1 in e2`, where e2 = ()
           *
-          *     we have already checked that c1 has a trivial precondition
-          *       and in most cases, c1 is some variant of PURE (top-level functions)
-          *     so if that's the case, we simply make cres as Tot unit
+          *     we have already checked that e1 has the right effect args
+          *     for it to be a top-level effect
           *
-          *     if c1 is not PURE, then we take a longer route and compute:
-          *       M.bind wp1 (fun _ -> (M.return_wp unit))
+          *     for wp effects that means trivial precondition,
+          *     and for indexed effects that means as per the top_level_effect
+          *     specification
           *
-          *     all this to remove the earlier usage of M.null_wp
+          *     Since the top-level effect is masked at this point,
+          *     we just return Tot unit and the final computation type
+          *
+          *     Note that for top-level lets, this cres is not used anyway
           *)
-         let cres =
```

Comment:
Trying to simplify computation of final computation type for a top-level `let x = e1 in ()`, making it always `Tot unit`. This final computation type is anyway not used, and `e1's` effect is already masked.